src/Sulu/Bundle/WebsiteBundle/Twig/Seo/SeoTwigExtension.php

```diff
@@ -168,16 +168,22 @@ private function renderAlternateLinks(array $urls, $webspaceKey)
     {
         $html = '';
 
-        $portal = $this->requestAnalyzer->getPortal();
         $defaultLocale = null;
+        $portal = $this->requestAnalyzer->getPortal();
         if ($portal) {
-            $defaultLocale = $portal->getDefaultLocalization()->getLocalization();
-
-            $html .= $this->renderAlternateLink($urls[$defaultLocale], $webspaceKey, $defaultLocale, true);
+            $defaultLocale = $portal->getXDefaultLocalization()->getLocalization();
         }
 
         foreach ($urls as $locale => $url) {
-            $html .= $this->renderAlternateLink($url, $webspaceKey, $locale);
+            // url = '/' means that there is no translation for this page
+            // the only exception is the homepage where the requested resource-locator is false
+            if ($url !== '/' || $this->requestAnalyzer->getResourceLocator() === false) {
```

Comment:
Isn't there a cleaner way to do this? Would be better to not return these links at all, instead of integrating this strange check here?